tests/utils/test_process_utils.py

```diff
@@ -0,0 +1,48 @@
+import os
+import tempfile
+import pytest
+
+from mlflow.utils.process import cache_return_value_per_process
+from multiprocessing import Process, Queue
+
+
+@cache_return_value_per_process
+def _gen_temp_dir1(_):
+    return tempfile.mkdtemp()
```

Comment:
I think we should return a random string here because `tempfile.mkdtemp` doesn't clean up the created experiment.

Reply:
Can we add a test the case where a function to decorate has no arguments?